jgrapht-core/src/main/java/org/jgrapht/alg/cycle/WeakChordalityInspector.java

```diff
@@ -0,0 +1,664 @@
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.util.Pair;
+import org.jgrapht.generate.ComplementGraphGenerator;
+import org.jgrapht.graph.AsUndirectedGraph;
+import org.jgrapht.graph.GraphWalk;
+import org.jgrapht.graph.Pseudograph;
+
+import java.util.*;
+
+/**
+ * Tests whether a graph is <a href="http://www.graphclasses.org/classes/gc_14.html">weakly chordal</a>.
+ * Weakly chordal graphs are also known as weakly triangulated graphs. Triangulated in the context of
+ * chordality has a different meaning than triangulated in the context of planarity, where it refers
+ * to a maximal planar graph, see: <a href="http://mathworld.wolfram.com/TriangulatedGraph.html">
+ * http://mathworld.wolfram.com/TriangulatedGraph.html</a>
+ * <p>
+ * The following definitions of  are equivalent:
+ * <ol>
+ * <li>
+ * A graph is weakly chordal (weakly triangulated) if neither it nor its complement
+ * contains a <a href="http://mathworld.wolfram.com/ChordlessCycle.html">chordless cycles</a> with
+ * five or more vertices.</li>
+ * <li> A 2-pair in a graph is a pair of non-adjacent vertices $x$, $y$ such that every chordless path has
+ * exactly two edges. A graph is weakly chordal if every connected
+ * <a href="https://en.wikipedia.org/wiki/Induced_subgraph">induced subgraph</a> $H$ that is not a complete
+ * graph, contains a 2-pair.</li>
+ * </ol>
+ * Chordal and weakly chordal graphs are <a href="http://mathworld.wolfram.com/PerfectGraph.html">perfect</a>.<br>
+ * For more details, refer to: Hayward, R.B. Weakly triangulated graphs, Journal of Combinatorial Theory, Series B,
+ * vol 39, Issue 3, pp 200-208, 1985.
+ * <p>
+ * The implementation in this class is based on: Lars Severin Skeide (2002)
+ * <a href="http://www.ii.uib.no/~skeide/rec_wcg.pdf"><i>Recognizing weakly chordal graphs</i></a>.
+ * Candidate Scientist Thesis in Informatics. Department of Informatics, University of Bergen, Norway.
+ * The terminology used in this implementation is consistent with the one used in this thesis.
+ * <p>
+ * Both the runtime complexity and space complexity of the algorithm implemented in this class is $\mathcal{O}(|E|^2)$.<br>
+ * The inspected {@code graph} is specified at the construction time and cannot be modified.
+ * When the graph is modified externally, the behavior of the {@code WeakChordalityInspector} is undefined.
+ * <p>
+ * In the case the inspected graph in not weakly chordal, this inspector provides a certificate in the form
+ * of some hole or anti-hole. The running time of finding a hole is $\mathcal{O}(|V| + |E|)$ and of
```

Comment:
To avoid confusion, let's link to the hole/anti-hole definitions on the graphclasses website, instead of to the wolfram website:
hole: http://graphclasses.org/smallgraphs.html#holes
anti-hole: http://graphclasses.org/smallgraphs.html#antiholes